packages/atomic/src/components/commerce/atomic-commerce-pager/atomic-commerce-pager.spec.ts

```diff
@@ -0,0 +1,325 @@
+import * as headless from '@coveo/headless/commerce';
+import {page} from '@vitest/browser/context';
+import '@vitest/browser/matchers.d.ts';
+import {html} from 'lit';
+import {describe, test, vi, expect, beforeEach} from 'vitest';
+import './atomic-commerce-pager';
+import {AtomicCommercePager} from './atomic-commerce-pager';
+
+vi.mock('@/src/components/common/interface/store', () => ({
+  createAppLoadedListener: vi.fn(),
+}));
+
+vi.mock('@/src/decorators/error-guard', () => ({
+  errorGuard: vi.fn(),
+}));
+
+vi.mock('@/src/decorators/bind-state', () => ({
+  bindStateToController: vi.fn(),
+}));
+
+const i18n = await createTestI18n();
+vi.mock('@/src/mixins/bindings-mixin', () => ({
+  InitializeBindingsMixin: vi.fn().mockImplementation((superClass) => {
+    return class extends superClass {
+      constructor(...args: unknown[]) {
+        super(...args);
+        this.bindings = {
+          store: {
+            state: {
+              resultList: {
+                focusOnFirstResultAfterNextSearch: () => {},
+              },
+            },
+          },
+          interfaceElement: {
+            type: 'product-listing',
+          },
+          i18n,
+        };
+        this.pagerState = {
+          page: 2,
+          totalPages: 10,
+          pageSize: 10,
+          totalEntries: 100,
+        };
+      }
+    };
+  }),
+  BindingController: class {},
+}));
+
+vi.mock('@coveo/headless/commerce', () => {
+  return {
+    buildProductListing: vi.fn(() => {
+      return {
+        pagination: vi.fn(() => ({
+          previousPage: vi.fn(),
+          selectPage: vi.fn(),
+          nextPage: vi.fn(),
+        })),
+      };
+    }),
+    buildSearch: vi.fn(() => {
+      return {
+        pagination: vi.fn(() => ({
+          previousPage: vi.fn(),
+          selectPage: vi.fn(),
+          nextPage: vi.fn(),
+        })),
+      };
+    }),
+  };
+});
+
+describe('AtomicCommercePager', () => {
+  let element: AtomicCommercePager;
+
+  const setupElement = async ({
+    numberOfPages,
+    mockedState = {
+      page: 0,
+      totalPages: 10,
+      pageSize: 10,
+      totalEntries: 100,
+    },
+  }: {numberOfPages?: number; mockedState?: PaginationState} = {}) => {
+    const element = await fixture<AtomicCommercePager>(
+      html`<atomic-commerce-pager></atomic-commerce-pager>`
+    );
+
+    if (numberOfPages !== undefined) {
+      element.setAttribute('number-of-pages', numberOfPages.toString());
+    }
+
+    element.pagerState = mockedState;
+    //@ts-expect-error - manually setting up this private property since a mocked implementation of createAppLoadedListener created problems. Couldn't manage keeping it to true when I needed it.
+    element.isAppLoaded = true;
+    element.initialize();
+
+    return element;
+  };
+
+  describe('when initializing', () => {
+    beforeEach(async () => {
+      element = await setupElement();
+    });
+
+    test('should validate props', () => {
```

Comment:
Rather than spying on a private method, you could spy on the @coveo/bueno Schema class and do something like:

```typescript
test('should validate props', () => {
  element.initialize();
  expect(schemaMock.mock.results[0].value.validate).toHaveBeenCalledWith({
    numberOfPages: Number.parseInt(element.getAttribute('number-of-pages')),
  });
});
```